test/edge.spec.ts

```diff
@@ -0,0 +1,118 @@
+import type { PrerenderManifest } from 'next/dist/build'
+
+const basePrerenderManifest: PrerenderManifest = {
+  version: 3,
+  routes: {},
+  dynamicRoutes: {},
+  notFoundRoutes: [],
+  preview: {
+    previewModeId: '',
+    previewModeSigningKey: '',
+    previewModeEncryptionKey: '',
+  },
+}
+
+describe('writeRscDataEdgeFunction', () => {
+  it('should return manifest entries for static appDir routes', async () => {
+    const prerenderManifest: PrerenderManifest = {
+      ...basePrerenderManifest,
+      routes: {
+        '/': {
+          initialRevalidateSeconds: false,
+          srcRoute: '/',
+          dataRoute: '/index.rsc',
+        },
+      },
+    }
+    const appPathRoutesManifest = {
+      '/page': '/',
+    }
+    const edgeManifest = await writeRscDataEdgeFunction({ prerenderManifest, appPathRoutesManifest })
+
+    expect(edgeManifest).toEqual([
+      {
+        function: 'rsc-data',
+        name: 'RSC data routing',
+        path: '/',
+      },
+      {
+        function: 'rsc-data',
+        name: 'RSC data routing',
+        path: '/index.rsc',
+      },
+    ])
+  })
+
+  it('should not return manifest entries for static appDir routes without dataRoutes', async () => {
+    const prerenderManifest: PrerenderManifest = {
+      ...basePrerenderManifest,
+      routes: {
+        '/api/hello': {
+          initialRevalidateSeconds: false,
+          srcRoute: '/api/hello',
+          dataRoute: null,
+        },
+      },
+    }
+    const appPathRoutesManifest = {
+      '/api/hello/route': '/api/hello',
+    }
+    const edgeManifest = await writeRscDataEdgeFunction({ prerenderManifest, appPathRoutesManifest })
+
+    expect(edgeManifest).toEqual([])
+  })
+
+  it('should return manifest entries for dynamic appDir routes', async () => {
+    const prerenderManifest: PrerenderManifest = {
+      ...basePrerenderManifest,
+      dynamicRoutes: {
+        '/blog/[author]': {
+          routeRegex: '^/blog/([^/]+?)(?:/)?$',
+          dataRoute: '/blog/[author].rsc',
+          fallback: null,
+          dataRouteRegex: '^/blog/([^/]+?)\\.rsc$',
+        },
+      },
+    }
+
+    const appPathRoutesManifest = {
+      '/blog/[author]/page': '/blog/[author]',
+    }
+    const edgeManifest = await writeRscDataEdgeFunction({ prerenderManifest, appPathRoutesManifest })
+
+    expect(edgeManifest).toEqual([
+      {
+        function: 'rsc-data',
+        name: 'RSC data routing',
+        pattern: '^/blog/([^/]+?)(?:/)?$',
+      },
+      {
+        function: 'rsc-data',
+        name: 'RSC data routing',
+        pattern: '^/blog/([^/]+?)\\.rsc$',
+      },
+    ])
+  })
+
+  it('should not return manifest entries for dynamic appDir routes without dataRoutes', async () => {
+    const prerenderManifest: PrerenderManifest = {
+      ...basePrerenderManifest,
+      dynamicRoutes: {
+        '/api/[endpoint]': {
+          routeRegex: '^/api/([^/]+?)(?:/)?$',
+          dataRoute: '/api/[endpoint].rsc',
```

Comment:
Confused here, as it looks like there is a dataRoute?

Reply:
`dynamicRoutes` (at least in our code) use `dataRouteRegex` field, not the `dataRoute`

Reply:
thanks folks, i've updated the test name to make it a bit clearer https://github.com/netlify/next-runtime/pull/2044/commits/b82d3d98c994fe10c787010d8ff00acf8a5326d0